### DISCUSSION TITLE
Bold on reciepts with addTextLine.

### DISCUSSION THREAD

**Discussion by sarmad-321** [tr3v3r/react-native-esc-pos-printer#168] [Q&A] [answered]
I want to bold in this situation like for example here i want to bold CheesBurger only can i achieve it? 
Also there is scenario where i want to show '(2) CheesBurger' so in this case i want "(2)" only bold .. is there any possibility for that ? kindly help trever.

await Printer.addTextLine(printerInstance, {
  left: 'Cheesburger',
  right: '3 EUR',
  gapSymbol: '_',
});

**Comment by tr3v3r** [comment]
@sarmad-321  could you try please to test the following commands:
```javascript
...

printerInstance.addTextStyle({
  em: PrinterConstants.TRUE,
});
printerInstance.addText('Hello');
printerInstance.addTextStyle({
  em: PrinterConstants.FALSE,
});
printerInstance.addText(' World!')
```

If it prints "Hello **World!**" in a one-line then the answer to your question will be **yes**. In this case I'll update addTextLine function

**Reply by sarmad-321** [reply]
Okay got your point ill test it out if it works like that and let you know

**Reply by sarmad-321** [reply]
Just tested it out and with this yeah it does works. so i think we can achieved this thing on addTextLine Function as well.
![WhatsApp Image 2024-08-07 at 4 32 58 PM](https://github.com/user-attachments/assets/eb1f4cc6-af5e-4541-ad1d-871f51677187)

**Reply by sarmad-321** [reply]
Also is it possible to some how use the smaller fontsize. smaller than 1 width & height. 
actually my client wants to see the products like this.  Product name x Price  . where price is very small in this 

![image](https://github.com/user-attachments/assets/608439e5-7ced-4f92-a4b1-377a22efcb54)

**Reply by tr3v3r** [reply]
You can make same test for font size :D

**Reply by sarmad-321** [reply]
i did by putting up the font size like this 
      await printerInstance.addTextSize({width: 0.5 , height:  0.5});
      turns out it didnt worked.

**Reply by tr3v3r** [reply]
> i did by putting up the font size like this await printerInstance.addTextSize({width: 0.5 , height: 0.5}); turns out it didnt worked.

that because 
<img width="786" alt="image" src="https://github.com/user-attachments/assets/b7d804ca-8f47-4b2c-ad1e-6163d742ee31">

**Reply by sarmad-321** [reply]
yeah and 1 is still very big to achieve what i need :(

**Reply by tr3v3r** [reply]
@sarmad-321 that is SDK limitation

**Comment by aargon007** [comment]
@sarmad-321 you can use esc pos encoder with this sdk. https://www.npmjs.com/package/esc-pos-encoder

**Comment by tr3v3r** [comment]
@sarmad-321 anyway I think you can not put less then "1"

**Comment by tr3v3r** [comment, marked as answer]
@sarmad-321 

Just released a new version with an alternative way of printing.
Please check [here](https://github.com/tr3v3r/react-native-esc-pos-printer/blob/main/docs/printer/Printer.md#printeraddviewshotprinterinstance-printer-params-addviewshotparams-promisevoid)

**Reply by sarmad-321** [reply]
Thank you tr3v3r for this new snapshot feature from the view. it will be very helpfull to those who have the same requirements as me. definitely gonna try this one. Once Again Thanks for this update.

**Reply by tr3v3r** [reply]
> Thank you tr3v3r for this new snapshot feature from the view. it will be very helpfull to those who have the same requirements as me. definitely gonna try this one. Once Again Thanks for this update. 

Looking forward to hearing from you regarding result
